x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/NativeUsersStore.java

```diff
@@ -459,24 +464,29 @@ public void onFailure(Exception e) {
     }
 
     public void deleteUser(final DeleteUserRequest deleteUserRequest, final ActionListener<Boolean> listener) {
-        securityIndex.prepareIndexIfNeededThenExecute(listener::onFailure, () -> {
-            DeleteRequest request = client.prepareDelete(SECURITY_INDEX_NAME,
+        final SecurityIndexManager frozenSecurityIndex = securityIndex.freeze();
+        if (frozenSecurityIndex.isAvailable() == false) {
+            listener.onFailure(frozenSecurityIndex.getUnavailableReason());
```

Comment:
The causes the behaviour of `DELETE _xpack/security/user/foo` to depend on whether the security index exists, which seems wrong. It's unnecessarily leaking internal implementation details in the API

```
# Index doesn't exist
DELETE _xpack/security/user/foo

{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","index_uuid":"_na_","index":".security"}],"type":"index_not_found_exception","reason":"no such index","index_uuid":"_na_","index":".security"},"status":404}

# Force index to be created
PUT _xpack/security/user/logstash_system/_disable

{}

# Try again, now the index does exist
DELETE _xpack/security/user/foo

{"found":false}
```